spec/features/admin/custom_fields/shared_custom_field_expectations.rb

```diff
@@ -0,0 +1,313 @@
+    it "does not show the enterprise upsell banner and has the 'Hierarchy' option for creation" do
+      expect(page).to have_no_text(I18n.t("ee.upsell.custom_field_hierarchies.description"))
+      cf_page.expect_having_create_item "Hierarchy"
+    end
+  end
+
+  context "with an active enterprise token without custom_field_hierarchies feature", with_ee: [:another_feature] do
+    it "shows the enterprise upsell banner and lacks the 'Hierarchy' option for creation" do
+      expect(page).to have_text(I18n.t("ee.upsell.custom_field_hierarchies.description"))
+      cf_page.expect_not_having_create_item "Hierarchy"
+    end
+  end
+
+  context "with a trial enterprise token", :with_ee_trial, with_ee: [:custom_field_hierarchies] do
+    it "shows the enterprise upsell banner and has the 'Hierarchy' option for creation" do
+      expect(page).to have_text(I18n.t("ee.upsell.custom_field_hierarchies.description"))
+      cf_page.expect_having_create_item "Hierarchy"
+    end
+  end
+end
+
+RSpec.shared_examples_for "expected fields for the custom field's format" do |type, format|
+  let(:cf_page) { Pages::CustomFields::IndexPage.new }
+  let(:user) { create(:admin) }
+
+  current_user { user }
+
+  before do
+    cf_page.visit_tab type
+  end
+
+  def expect_page_to_have_texts(*text)
+    text.each do |t|
+      expect(page).to have_text(t)
+    end
+  end
+
+  def expect_page_not_to_have_texts(*text)
+    text.each do |t|
+      expect(page).to have_no_text(t)
+    end
+  end
+
+  # Form element labels, default English translation in the trailing comment:
+  let(:label_min_length) { I18n.t("activerecord.attributes.custom_field.min_length") } # Minimum length
+  let(:label_max_length) { I18n.t("activerecord.attributes.custom_field.max_length") } # Maximum length
+  let(:label_regexp) { I18n.t("activerecord.attributes.custom_field.regexp") } # Regular expression
+  let(:label_multi_value) { I18n.t("activerecord.attributes.custom_field.multi_value") } # Allow multi-select
+  # Allow non-open versions
+  let(:label_allow_non_open_versions) do
+    I18n.t("activerecord.attributes.custom_field.allow_non_open_versions")
+  end
+  # Possible values, capitalized on UI
+  let(:label_possible_values) do
+    I18n.t("activerecord.attributes.custom_field.possible_values").upcase
+  end
+  let(:label_default_value) { I18n.t("activerecord.attributes.custom_field.default_value") } # Default value
+  let(:label_is_required) { I18n.t("activerecord.attributes.custom_field.is_required") } # Required
+  let(:label_ee_banner_hierarchy) { I18n.t("ee.upsell.custom_field_hierarchies.description") } # Hierarchy Enterprise banner
+  # Spent time SFs don't show "Searchable". Not tested here.
+  # Project CFs don't show "For all projects" and "Used as a filter". Not tested here.
+  # Content right to left is not shown for Project CFs Long text. Strange. Not tested.
```

Comment:
That is indeed strange and seems to be intended. The attribute is [explicitly only added for work package custom fields](https://github.com/opf/openproject/blob/28b52bbb4f3de396d6e39d7167aad82e9aaecb26/app/views/custom_fields/_form.html.erb#L190). I too do not understand why a project text field would be different from a work package text field.

After some digging it seems that this was only requested for work package custom fields due to a [customer requirement](https://community.openproject.org/wp/31656).

In any case, supporting this attribute for ProjectCustomFields is outside of the scope of this PR :) So for the purposes of this PR, we are good 👍